src/drawing/__tests__/DrawingCreator-test.tsx

```diff
@@ -0,0 +1,159 @@
+        shallow(<DrawingCreator {...getDefaults()} {...props} />);
+    const setDrawingStatus = jest.fn();
+    let capturedPointsRef: MutableRefObject<Array<Position>> = { current: [] };
+    let creatorRef = { current: creatorEl };
+    let drawignDirtyRef = { current: false };
+    let drawingPathRef: MutableRefObject<DrawingPathRef | null> = { current: null };
+    let drawingSVGRef = { current: null };
+    let renderHandlerRef = { current: null };
+
+    beforeEach(() => {
+        capturedPointsRef = { current: [] };
+        creatorRef = { current: creatorEl };
+        drawignDirtyRef = { current: false };
+        drawingPathRef = { current: null };
+        drawingSVGRef = { current: null };
+        renderHandlerRef = { current: null };
+
+        jest.spyOn(React, 'useState').mockImplementation(() => [Status.init, setDrawingStatus]);
+        jest.spyOn(React, 'useEffect').mockImplementation(noop);
+        jest.spyOn(React, 'useRef')
+            .mockImplementation(() => React.createRef())
+            .mockReturnValueOnce(capturedPointsRef)
+            .mockReturnValueOnce(creatorRef)
+            .mockReturnValueOnce(drawignDirtyRef)
+            .mockReturnValueOnce(drawingPathRef)
+            .mockReturnValueOnce(drawingSVGRef)
+            .mockReturnValueOnce(renderHandlerRef);
+    });
+
+    describe('render', () => {
+        test('should render PointerCapture', () => {
+            const wrapper = getWrapper();
+
+            expect(wrapper.find(PointerCapture).exists()).toBe(true);
+            expect(wrapper.find(PointerCapture).prop('className')).toBe('ba-DrawingCreator');
+        });
+
+        test.each`
+            status             | exists
+            ${Status.init}     | ${false}
+            ${Status.dragging} | ${false}
+            ${Status.drawing}  | ${true}
+        `('should render DrawingSVG if status=$status? $exists', ({ status, exists }) => {
+            jest.spyOn(React, 'useState').mockImplementationOnce(() => [status, setDrawingStatus]);
+            const wrapper = getWrapper();
+
+            expect(wrapper.find(PointerCapture).exists()).toBe(true);
+            expect(wrapper.find(DrawingSVG).exists()).toBe(exists);
+        });
+    });
+
+    describe('drawing', () => {
+        test('should trigger start logic on draw start', () => {
+            const wrapper = getWrapper();
+
+            wrapper.find(PointerCapture).prop('onDrawStart')(1, 2);
+
+            expect(setDrawingStatus).toHaveBeenCalledWith(Status.dragging);
+        });
+
+        test('should invoke onStart callback on draw update', () => {
+            const onStart = jest.fn();
+            const wrapper = getWrapper({ onStart });
+
+            wrapper.find(PointerCapture).prop('onDrawUpdate')(2, 3);
+
+            expect(setDrawingStatus).toHaveBeenCalledWith(Status.drawing);
+            expect(capturedPointsRef.current).toEqual([{ x: 2, y: 3 }]);
```

Comment:
We should try to test only the output of the component rather than its internal state.